mlflow/utils/environment.py

```diff
@@ -500,6 +501,19 @@ def _process_pip_requirements(
     return conda_env, pip_reqs, constraints
 
 
+def _check_for_duplicate_requirements(requirements):
+    """
+    Checks if duplicate base package requirements are specified in any list of requirements
+    and returns the list of duplicate base package names
+    """
+    base_package_names = [
+        Requirement(package).name for package in requirements if not package.startswith("-")
```

Comment:
Does `Requirement(package)` work for requirements like `git+https://github.com/...`? I remember it doesn't.

Reply:
> Does `Requirement(package)` work for requirements like `git+https://github.com/...`? I remember it doesn't. If it doesn't, we need to suppress an exception.

Reply:
Updated with handling and updated the test with all supported methods of installing packages (or restricting installation) in python as of the most recent version of pip.